dev-client/src/auth/index.ts

```diff
@@ -0,0 +1,46 @@
+import Config from 'react-native-config';
+import {OAuthProvider} from '../types';
+
+function checkFields(fields: string[], json: object) {
+  return fields.filter(key => !Object.hasOwn(json, key)).length > 0;
+}
+
+export async function exchangeToken(
+  identityJwt: string,
+  provider: OAuthProvider,
+) {
+  const resp = await fetch(Config.TERRASO_BACKEND + '/auth/token-exchange', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+    body: JSON.stringify({
+      provider,
+      jwt: identityJwt,
+    }),
+  });
+  if (!resp.ok) {
+    // TODO: handle error
+    throw 'Error with token exchange';
+  }
+  const payload = await resp.json();
+
+  if (
+    checkFields(['atoken', 'rtoken', 'user'], payload) ||
+    checkFields(['email', 'firstName'], payload.user)
+  ) {
+    // TODO: handle error
+    console.error(payload);
+    throw 'Bad token JSON';
+  }
+
+  // TODO: Just doing this to get the right types, has to be a better way
```

Comment:
suggestion: you can give the type of the request JSON as a type argument to the `request` method i mentioned above. unfortunately there is a longstanding missing feature of TypeScript to validate that a runtime value conforms to a TypeScript type, so it won't actually _check_ the fields, but it at least lets TypeScript know what they're supposed to be